isis/tests/FunctionalTestsIsisImportEis.cpp

```diff
@@ -0,0 +1,579 @@
+#include <QTemporaryFile>
+#include <QFileInfo>
+#include <QTextStream>
+#include <QByteArray>
+#include <QDataStream>
+
+#include <nlohmann/json.hpp>
+#include "Fixtures.h"
+#include "Histogram.h"
+#include "md5wrapper.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "PvlKeyword.h"
+#include "TestUtilities.h"
+#include "isisimport.h"
+#include "gmock/gmock.h"
+
+using namespace Isis;
+using namespace testing;
+using json = nlohmann::json;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/isisimport.xml").expanded();
+
+TEST_F(TempTestingFiles, FunctionalTestIsisImportEisNacFrame){
+  std::istringstream PvlInput(R"(
+    Object = IsisCube
+      Object = Core
+        StartByte   = 65537
+        Format      = Tile
+        TileSamples = 445
+        TileLines   = 958
+
+        Group = Dimensions
+          Samples = 1335
+          Lines   = 3832
+          Bands   = 1
+        End_Group
+
+        Group = Pixels
+          Type       = SignedWord
+          ByteOrder  = Lsb
+          Base       = 0.0
+          Multiplier = 1.0
+        End_Group
+      End_Object
+
+      Group = Instrument
+        SpacecraftName   = "Europa Clipper"
+        InstrumentId     = "EIS NAC PB"
+        TargetName       = Didymos
+        StartTime        = 2021-03-06T04:05:27.77
+        ExposureDuration = 1.0 <seconds>
+      End_Group
+
+      Group = BandBin
+        FilterName = CLEAR
+        Center     = 702.5 <nm>
+        Width      = 695 <nm>
+      End_Group
+
+      Group = Kernels
+        NaifFrameCode = -159103
+      End_Group
+    End_Object
+
+    Object = Label
+      Bytes = 65536
+    End_Object
+
+    Object = History
+      Name      = IsisCube
+      StartByte = 10296977
+      Bytes     = 657
+    End_Object
+
+    Object = OriginalXmlLabel
+      Name      = IsisCube
+      StartByte = 10297634
+      Bytes     = 13238
+      ByteOrder = Lsb
+    End_Object
+    End
+  )");
+  QString dataFilePath= "data/isisimport/eispds/nacFrame/nac000xxx_2022145t000000_0000000001_frame_raw02.xml";
+  QString dataFileName = "nac000xxx_2022145t000000_0000000001_frame_raw02.xml";
+  QString imageFileName = "nac000xxx_2022145t000000_0000000001_frame_raw02.img";
+  QString templateFile = "../appdata/import/PDS4/ClipperEIS.tpl";
+  QString cubeFileName = tempDir.path() + "/nacFrame.cub";
+
+  int samples = 1335;
+  int lines = 3832;
+  int bytes = 2;
+
+  // create a temp img file and write data to it
+  QFile tempImgFile(tempDir.path() + "/" + imageFileName);;
+
+  (tempDir.path() + "/" + imageFileName);
+  if(!tempImgFile.open(QFile::WriteOnly | QFile::Text)){
+      FAIL() << " Could not open file for writing";
+  }
+  QTextStream out(&tempImgFile);
+
+  // generate lines 
+  QByteArray writeToFile = QByteArray();
+  short int fill = 0;
+  for(int i=-1; i<(samples * bytes); i++){
+    writeToFile.append(fill);
+  }
+
+  // write the lines to the temp file 
+  for(int i=0; i<lines; i++){
+    QTextStream out(&tempImgFile);
```

Comment:
Change to a QDataStream and you won't need to fill a ByteArray. You can just pipe `fill` straight into the stream and it will go to the file.

Right now, this is getting converted to text https://doc.qt.io/qt-6/qtextstream.html#operator-lt-lt-15